sale_report_margin/reports/sale_report.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleReport(models.Model):
+    _inherit = "sale.report"
+
+    purchase_price = fields.Float(
+        string='Purchase Price',
+        readonly=True,
+    )
+
+    def _select(self):
```

Comment:
This has been modified upstream. The hint about this is the coverage drop. Now everything goes in one method:

https://github.com/odoo/odoo/blob/a7d24cabafe1a9c71f9dbdb7cf3f129006250951/addons/sale/report/sale_report.py#L58

and in this case it's simple, as you only have to add the field in the argument:

```python
def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
    fields['purchase_price'] = ',\nSUM(l.purchase_price / CASE COALESCE(s.currency_rate, 0)) as purchase_price'
    return super()._query(self, with_clause=with_clause, fields=fields, groupby=groupby, from_clause=from_clause)
```
My only doubt (but it's an Odoo problem) is if the argument `field={}` will have problem in a multi-DB (https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python).